pages/api/user/index.ts

```diff
@@ -0,0 +1,180 @@
+import { PrismaClient } from '@prisma/client';
+import type { NextApiRequest, NextApiResponse } from 'next';
+import { ApiResponse, EmptyPayload, StatusMessageType } from '../../../types/apiResponse';
+import { Nullable } from '../../../types/utils';
+import { getUserFromJwt, UserDetailsFromRequest, withAuthUser } from '../../../utils/auth/jwtHelpers';
+import { createJsonResponse, HttpMethod, HttpStatus, rejectHttpMethod } from '../../../utils/http/httpHelpers';
+import { withPrismaErrorHandling } from '../../../utils/prisma/prismaHelpers';
+
+/** Limited set of data representing user and the list of user application IDs. Only for use in this endpoint. */
+interface UserData {
+  uid: string;
+  applications: { id: number }[];
+}
+
+enum LinkAccountResult {
+  OLD_USER_DELETED,
+  NEW_USER_HAS_DATA,
+  NEW_USER_LINKED,
+}
+
+const { ERROR, INFORMATION, SUCCESS } = StatusMessageType;
+const prisma = new PrismaClient();
+
+async function handler(currentUid: string, req: NextApiRequest, res: NextApiResponse<ApiResponse<EmptyPayload>>) {
+  switch (req.method) {
+    case HttpMethod.POST:
```

Comment:
Lets have a `handlePost` method and move the logic in the switch statement there.

Reply:
Is this a nit or is it important to you? Cuz I don't see how that is clearer, extracting these few lines into another function makes things harder to follow IMO. A bit of logic in the switch case makes the different handler flows obvious.

Reply:
Its a nit. This method seemed weird because the abstraction level is different for this switch case (according to SLAP at least).

Reply:
Hmm fair..